check_patroni/cluster.py

```diff
@@ -52,19 +52,44 @@ def probe(self) -> Iterable[nagiosplugin.Metric]:
         item_dict = self.rest_api("cluster")
 
         is_leader_found = False
+        is_standby_leader_found = False
+        is_standby_leader_in_arc_rec = False
         for member in item_dict["members"]:
-            if (
-                member["role"] in ("leader", "standby_leader")
-                and member["state"] == "running"
-            ):
+            if member["role"] == "leader" and member["state"] == "running":
                 is_leader_found = True
                 break
 
+            if member["role"] == "standby_leader":
+                if member["state"] not in ["streaming", "in archive recovery"]:
+                    # for patroni >= 3.0.4 any state would be wrong
+                    # for patroni <  3.0.4 a state different from running would be wrong
+                    if self.has_detailed_states() or (
+                        not self.has_detailed_states() and member["state"] != "running"
```

Comment:
```suggestion
                    if self.has_detailed_states() or member["state"] != "running":
```
isn't this enough?

Reply:
Yes (sight, I had to draw a boolean table ...)